engine/configurator/conf_main.c

```diff
@@ -3023,12 +2950,12 @@ main(int argc, char **argv)
     }
 
     if ((filename = (char *)malloc(strlen(tmp_dir) +
-                                   strlen("/te_cfg_tmp.xml") + 1)) == NULL)
+                                   strlen("/te_cfg_tmp.yml") + 1)) == NULL)
     {
         ERROR("No enough memory");
         goto exit;
     }
-    sprintf(filename, "%s/te_cfg_tmp.xml", tmp_dir);
+    sprintf(filename, "%s/te_cfg_tmp.yml", tmp_dir);
```

Comment:
Consider making a temporary file in a more sensible way.

Reply:
It is definitely unrelated changes. I've added one more commit to make it.